Dockerfile

```diff
@@ -1,11 +1,33 @@
-FROM openjdk:11-jdk as builder
+###############################
+# Maven builder
+###############################
+FROM adoptopenjdk/openjdk11:jdk-11.0.6_10-alpine-slim as builder
 WORKDIR /usr/src/app
-ADD . .
-RUN ./mvnw clean package
-
-FROM openjdk:11-jre-slim
-COPY --from=builder /usr/src/app/target/workflow-relay-*.jar /usr/bin/workflow-relay.jar
-RUN adduser --disabled-password --disabled-login --quiet --gecos '' relay
-USER relay
-CMD ["java", "-ea", "-jar", "/usr/bin/workflow-relay.jar"]
+COPY . .
+RUN ./mvnw clean package -DskipTests
+
+##############################
+# Server
+##############################
+FROM adoptopenjdk/openjdk11:jre-11.0.6_10-alpine as server
+
+# Paths
+ENV APP_HOME /usr/src/app
+ENV JAR_FILE $APP_HOME/workflow-relay.jar
+ENV APP_USER wfuser
+ENV APP_UID 9999
+ENV APP_GID 9999
+
+RUN addgroup -S -g $APP_GID $APP_USER  \
+    && adduser -S -u $APP_UID -G $APP_USER $APP_USER  \
+    && mkdir -p $APP_HOME \
+    && chown -R $APP_UID:$APP_GID $APP_HOME
+
+COPY --from=builder /usr/src/app/target/workflow-relay-*.jar $JAR_FILE
+
+USER $APP_UID
+
+WORKDIR $APP_HOME
+
+CMD java -ea -jar $JAR_FILE
 EXPOSE 8080/tcp
```

Comment:
I feel like this either needs to be configurable or its not necessary at all, want to look into it as part of this?

Reply:
there is not much practical significance. Its mostly to indicate to the developer which port the app is served on. It could be useful with the `docker run -P` command, which published ALL exposed ports to random ports.